### ISSUE TITLE
In error messages, name the method that we can't intercept on an instance

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#786]
Inspired by a [StackOverflow question](http://stackoverflow.com/q/38152748/1199).

The questioner had the equivalent of

``` c#
public interface IHaveAList
{
    IEnumerable<string> GetList();
}

var fake = A.Fake<IHaveAList>();
A.CallTo(() => fake.GetList().Single())
    .Throws(new Exception());
```

And didn't understand why he was told 

```
FakeItEasy.Configuration.FakeConfigurationException: 
The current proxy generator can not intercept the specified method for the following reason: 
- Extension methods can not be intercepted since they're static.
```

The thought was the complaint was about `GetList`, when really it was about `Single`.

We should say which method (I'd almost hope for something like `Enumerable.Single<TSource>(this IEnumerable<TSource> source)`, but whatever we can get would be great.
We could maybe take a cue from `DefaultFakeObjectCallFormatter`, which has very advanced formatting rules based on whether the method is a method or a property and so on, but it's concerned with object calls and we only have `MethodInfo`s available where we build the exception message above (in `DefaultInterceptionAsserter.AssertThatMethodCanBeInterceptedOnInstance`), so we'd need to do a little work for  a full-fidelity solution.

**Comment by adamralph** [comment]
Makes sense to me.

**Comment by thomaslevesque** [comment]
> (I'd almost hope for something like `Enumerable.Single<TSource>(this IEnumerable<TSource> source)`

We should probably use the CLR format instead (see #1047)

**Comment by thomaslevesque** [comment]
btw @blairconrad did you intend to omit the namespace? Should it be `System.Linq.Enumerable.Single<TSource>(this System.Collections.Generic.IEnumerable<TSource> source)` instead? It's a bit verbose (especially when `TSource` will be replaced with an actual type), but it's more specific.

**Comment by thomaslevesque** [comment]
Also, if we're going to change this message, what do you think about dropping the [two blank lines and two spaces at the beginning of the message](https://github.com/FakeItEasy/FakeItEasy/blob/79e46300d74c13558446b74f9d80ba2418c721df/src/FakeItEasy/Configuration/DefaultInterceptionAsserter.cs#L23-L25)? As far as I can tell, they don't bring any benefit.

**Comment by blairconrad** [comment]
@thomaslevesque, as you say, we should use the CLR format. And namespaces are good.

The whitespace is there to make the exception render pretty when the user makes this error:

```
FakeConfigurationException: 

  The current proxy generator can not intercept the specified method for the following reason:
    - Non virtual methods can not be intercepted.
```

(You probably know this, and were still thinking they could be dropped.)

I think this sort of thing is done in multiple places, and if we were going to start reformatting messages, it might be better to take a look at all of them?
(Which I'm not that excited about…)

**Comment by thomaslevesque** [comment]
> (You probably know this, and were still thinking they could be dropped.)

Actually, I didn't realize this. I guess it makes sense, then.

**Comment by blairconrad** [comment]
This issue was fixed in [Release 3.3.0](https://github.com/FakeItEasy/FakeItEasy/releases/3.3.0).
